llvmlite/binding/newpassmangers.py

```diff
@@ -0,0 +1,305 @@
+
+    def __init__(self, ptr=None):
+        if ptr is None:
+            ptr = ffi.lib.LLVMPY_CreatePipelineTuningOptions()
+            self._opt_level = 2
+            self._size_level = 0
+        ffi.ObjectRef.__init__(self, ptr)
+
+    @property
+    def loop_interleaving(self):
+        return ffi.lib.LLVMPY_PTOGetLoopInterleaving(self)
+
+    @loop_interleaving.setter
+    def loop_interleaving(self, value):
+        ffi.lib.LLVMPY_PTOSetLoopInterleaving(self, value)
+
+    @property
+    def loop_vectorization(self):
+        return ffi.lib.LLVMPY_PTOGetLoopVectorization(self)
+
+    @loop_vectorization.setter
+    def loop_vectorization(self, value):
+        ffi.lib.LLVMPY_PTOSetLoopVectorization(self, value)
+
+    @property
+    def slp_vectorization(self):
+        return ffi.lib.LLVMPY_PTOGetSLPVectorization(self)
+
+    @slp_vectorization.setter
+    def slp_vectorization(self, value):
+        ffi.lib.LLVMPY_PTOSetSLPVectorization(self, value)
+
+    @property
+    def loop_unrolling(self):
+        return ffi.lib.LLVMPY_PTOGetLoopUnrolling(self)
+
+    @loop_unrolling.setter
+    def loop_unrolling(self, value):
+        ffi.lib.LLVMPY_PTOSetLoopUnrolling(self, value)
+
+    # // FIXME: Available from llvm16
+    # @property
+    # def inlining_threshold(self):
+    #     return ffi.lib.LLVMPY_PTOGetInlinerThreshold(self)
+
+    # @inlining_threshold.setter
+    # def inlining_threshold(self, value):
+    #     ffi.lib.LLVMPY_PTOSetInlinerThreshold(self, value)
+
+    # Not part of PTO
+    @property
+    def opt_level(self):
+        """
+        The general optimization level as an integer between 0 and 3.
+        """
+        return self._opt_level
+
+    @opt_level.setter
+    def opt_level(self, level):
+        self._opt_level = level
+
+    @property
+    def size_level(self):
+        """
+        Whether and how much to optimize for size.
+        An integer between 0 and 2.
+        """
+        return self._size_level
+
+    @size_level.setter
+    def size_level(self, size):
+        self._size_level = size
+
+    def _dispose(self):
+        ffi.lib.LLVMPY_DisposePipelineTuningOptions(self)
+
+
+class PassBuilder(ffi.ObjectRef):
+
+    def __init__(self, tm, pto, ptr=None):
+        if ptr is None:
+            self._pto = pto
+            self._tm = tm
+            ptr = ffi.lib.LLVMPY_CreatePassBuilder(tm, pto)
+        ffi.ObjectRef.__init__(self, ptr)
+
+    def getNewModulePassManager(self):
+        return NewModulePassManager(
+            ffi.lib.LLVMPY_buildPerModuleDefaultPipeline(
+                self, self._pto.opt_level, self._pto.size_level)
+        )
+
+    def getNewFunctionPassManager(self):
+        return NewFunctionPassManger(
+            ffi.lib.LLVMPY_buildFunctionSimplificationPipeline(
+                self, self._pto.opt_level, self._pto.size_level)
+        )
+
+    def _dispose(self):
+        # Should I explicitly delete pointer to pto and tm?
```

Comment:
No, they will be disposed when they are collected by the GC after they're out of scope everywhere.